File: nordeus_challenge_backend_spring_boot/src/main/java/com/mmileticc/nordeuschallenge/model/dto/BattleStateDTO.java
package com.mmileticc.nordeuschallenge.model.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class BattleStateDTO {

    private Long heroId;
    private Long monsterId;
    private Long selectedMoveId;
    private Integer heroMaxHealth;
    private List<Long> heroMoveIds = new ArrayList<>();

    private Integer heroCurrentHealth;
    private Integer heroAttack;
    private Integer heroDefense;
    private Integer heroMagic;

    private Integer monsterCurrentHealth;
    private Integer monsterAttack;
    private Integer monsterDefense;
    private Integer monsterMagic;

    private List<ActiveBuffDTO> activeBuffs = new ArrayList<>();
}

File: Gauntlet of the Hero - GOTH/Assets/_Scripts/Models/Backend/BattleStateDto.cs
using System.Collections.Generic;
using Newtonsoft.Json;

namespace GauntletOfTheHero.Backend.Models
{
    [System.Serializable]
    public class BattleStateDto
    {
        [JsonProperty("heroId")]
        public long heroId;

        [JsonProperty("monsterId")]
        public long monsterId;

        [JsonProperty("selectedMoveId")]
        public long? selectedMoveId;

        [JsonProperty("heroMaxHealth")]
        public int? heroMaxHealth;

        [JsonProperty("heroMoveIds")]
        public List<long> heroMoveIds = new List<long>();

        [JsonProperty("heroCurrentHealth")]
        public int? heroCurrentHealth;

        [JsonProperty("heroAttack")]
        public int? heroAttack;

        [JsonProperty("heroDefense")]
        public int? heroDefense;

        [JsonProperty("heroMagic")]
        public int? heroMagic;

        [JsonProperty("monsterCurrentHealth")]
        public int? monsterCurrentHealth;

        [JsonProperty("monsterAttack")]
        public int? monsterAttack;

        [JsonProperty("monsterDefense")]
        public int? monsterDefense;

        [JsonProperty("monsterMagic")]
        public int? monsterMagic;

        [JsonProperty("activeBuffs")]
        public List<ActiveBuffDto> activeBuffs = new List<ActiveBuffDto>();
    }
}